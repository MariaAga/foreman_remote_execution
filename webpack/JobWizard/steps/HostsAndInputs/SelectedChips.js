import React from 'react';
import PropTypes from 'prop-types';
import { Chip, ChipGroup, Button } from '@patternfly/react-core';
import { translate as __ } from 'foremanReact/common/I18n';
import { hostMethods } from '../../JobWizardConstants';

const SelectedChip = ({ selected, setSelected, categoryName }) => {
  const deleteItem = itemToRemove => {
    setSelected(oldSelected =>
      oldSelected.filter(item => item !== itemToRemove)
    );
  };
  return (
    <ChipGroup className="hosts-chip-group" categoryName={categoryName}>
      {selected.map(chip => (
        <Chip key={chip} id={chip} onClick={() => deleteItem(chip)}>
          {chip}
        </Chip>
      ))}
    </ChipGroup>
  );
};

export const SelectedChips = ({
  selectedHosts,
  setSelectedHosts,
  selectedHostCollections,
  setSelectedHostCollections,
  selectedHostGroups,
  setSelectedHostGroups,
}) => (
  <div className="selected-chips">
    <SelectedChip
      selected={selectedHosts}
      categoryName={hostMethods.hosts}
      setSelected={setSelectedHosts}
    />
    <SelectedChip
      selected={selectedHostCollections}
      categoryName={hostMethods.hostCollections}
      setSelected={setSelectedHostCollections}
    />

    <SelectedChip
      selected={selectedHostGroups}
      categoryName={hostMethods.hostGroups}
      setSelected={setSelectedHostGroups}
    />
    <Button variant="link">{__('Clear filters')}</Button>
  </div>
);

SelectedChips.propTypes = {
  selectedHosts: PropTypes.array.isRequired,
  setSelectedHosts: PropTypes.func.isRequired,
  selectedHostCollections: PropTypes.array.isRequired,
  setSelectedHostCollections: PropTypes.func.isRequired,
  selectedHostGroups: PropTypes.array.isRequired,
  setSelectedHostGroups: PropTypes.func.isRequired,
};

SelectedChip.propTypes = {
  categoryName: PropTypes.string.isRequired,
  selected: PropTypes.array.isRequired,
  setSelected: PropTypes.func.isRequired,
};
