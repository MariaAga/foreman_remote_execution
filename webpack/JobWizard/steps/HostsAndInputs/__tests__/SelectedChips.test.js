import React from 'react';
import { testComponentSnapshotsWithFixtures } from '@theforeman/test';
import { render, fireEvent, screen } from '@testing-library/react';
import * as patternfly from '@patternfly/react-core';
import { SelectedChips } from '../SelectedChips';

jest.spyOn(patternfly, 'Chip');
jest.spyOn(patternfly, 'ChipGroup');
patternfly.Chip.mockImplementation(props => <div {...props} />);
patternfly.ChipGroup.mockImplementation(props => <div {...props} />);

const selectedHosts = ['host1', 'host2', 'host3'];
const setSelectedHosts = jest.fn();
const props = {
  selectedHosts,
  setSelectedHosts,
  selectedHostCollections: [],
  setSelectedHostCollections: jest.fn(),
  selectedHostGroups: [],
  setSelectedHostGroups: jest.fn(),
};
const fixtures = {
  'renders with props': props,
};

describe('SelectedChips', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(SelectedChips, fixtures));
  it('deleting', () => {
    patternfly.Chip.mockRestore();
    patternfly.ChipGroup.mockRestore();
    render(<SelectedChips {...props} />);
    const chip1 = screen.getByLabelText('host1', { selector: 'button' });
    fireEvent.click(chip1);
    expect(setSelectedHosts).toBeCalled();
  });
});
