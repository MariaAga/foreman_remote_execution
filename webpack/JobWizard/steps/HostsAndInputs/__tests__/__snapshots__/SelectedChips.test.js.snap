// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedChips rendering renders with props 1`] = `
<div
  className="selected-chips"
>
  <SelectedChip
    categoryName="Hosts"
    selected={
      Array [
        "host1",
        "host2",
        "host3",
      ]
    }
    setSelected={[MockFunction]}
  />
  <SelectedChip
    categoryName="Host collections"
    selected={Array []}
    setSelected={[MockFunction]}
  />
  <SelectedChip
    categoryName="Host groups"
    selected={Array []}
    setSelected={[MockFunction]}
  />
  <Button
    variant="link"
  >
    Clear filters
  </Button>
</div>
`;
